airflow/www/views.py

```diff
@@ -3727,8 +3727,41 @@ def conf(self):
         raw = request.args.get('raw') == "true"
         title = "Airflow Configuration"
         subtitle = AIRFLOW_CONFIG
+
+        expose_config = conf.get('webserver', 'expose_config')
+
         # Don't show config when expose_config variable is False in airflow config
-        if conf.getboolean("webserver", "expose_config"):
+        # Don't show sensitive config values if expose_config variable is 'non-sensitive-only'
+        # in airflow config
+        if expose_config.lower() == 'non-sensitive-only':
+            from airflow.configuration import SENSITIVE_CONFIG_VALUES
+
+            with open(AIRFLOW_CONFIG) as file:
+                config = file.readlines()
+                for line in config:
+                    for _, key in SENSITIVE_CONFIG_VALUES:
+                        # this masks the keys wherever it's found not
+                        # minding the section
+                        if key in line and not line.startswith('#'):
+                            config[config.index(line)] = key + ' = ***\n'
+                            break
+
+                config = ''.join(config)
+
+            running_conf = conf.as_dict(True, True)
+            for section, key in SENSITIVE_CONFIG_VALUES:
+                running_conf_value = running_conf[section].get(key, None)
+                if running_conf_value:
+                    new = ('***', running_conf_value[1])
+                    running_conf[section][key] = new
```

Comment:
I wonder if this behaviour should live as an option inside conf as_dict

Reply:
Yeah, found out that the `display_sensitive` option was made to hide sensitive fields but was not hidden those that come from airflow.cfg in the running config. Fixed now